### PR TITLE
add maven release

### PR DESCRIPTION
**Pull request by hupling**
**Description**

Short description or comments

**Reference**

Issues https://github.com/it-at-m/.github/issues/43

## Summary by CodeRabbit

- **New Features**
	- Introduced a new automated workflow for releasing Maven applications, including building Docker images and creating GitHub releases.
	- Users can manually trigger the release process with specified version inputs and application paths.

### CODE REVIEW

File: .github/workflows/release-maven-image.yaml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        description: "Default version to use for new local working copy."
+        required: true
+        default: "X.Y.Z-SNAPSHOT"
+      app-path:
+        type: choice
+        description: Service-Name
+        required: true
+        options:
+          - sps-backend # todo: muss angepasst werden. eigenen Ordner verwenden
+          - sps-frontend # todo: muss angepasst werden. eigenen Ordner verwenden
```
Comment (hupling) [inline .github/workflows/release-maven-image.yaml:20]:
```suggestion
          - refarch-backend
          - refarch-eai
```

File: .github/workflows/release-maven-image.yaml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            type=semver,pattern={{version}},value=${{ github.event.inputs.releaseVersion }}
+
+  create-github-release:
+    needs: release-maven
+    runs-on: ubuntu-latest
+    steps:
+      - name: Create GitHub Release
+        id: create_release
+        uses: softprops/action-gh-release@v2
+        with:
```
Comment (hupling) [inline .github/workflows/release-maven-image.yaml:61]: hier fehlt noch 
```yml
       files: $app-path/target/${jar-name}.jar
```

**Comment by hupling** [comment]
@it-at-m/sps  zur Info

File: .github/workflows/release-maven-image.yaml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    env:
+      TZ: Europe/Berlin # timezone
+    steps:
+      - id: maven-release-step
+        uses: it-at-m/.github/.github/actions/action-maven-release@main
+        with:
+          app-path: "/${{ github.event.inputs.app-path }}"
+          releaseVersion: "${{ github.event.inputs.releaseVersion }}"
+          developmentVersion: "${{ github.event.inputs.developmentVersion }}"
+          java-version: 17
```
Comment (simonhir) [inline .github/workflows/release-maven-image.yaml:37]:
```suggestion
          java-version: 21
```

Comment (hupling) [reply .github/workflows/release-maven-image.yaml:37]: oder zeile einfach komplett raus werfen. und nur in den templates das default angeben

File: .github/workflows/release-maven-image.yaml
Comment (devtobi) [inline .github/workflows/release-maven-image.yaml:1]: I don't really understand why the creation of a maven release and the creation of a docker image have to be coupled together in this workflow. Is there any reason for doing so?
If yes it is definitely better to put this into a reusable workflow in the `.github` repository and thus simplify the CI/CD code in the individual repositories even more.

**Review by devtobi — DISMISSED** [review]

File: .github/workflows/release-maven-image.yaml [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    runs-on: ubuntu-latest
+    needs: release-maven
+    steps:
+      - uses: it-at-m/.github/.github/actions/action-build-image@main
+        with:
+          path: "${{ github.event.inputs.app-path }}"
+          image-name: "${{ github.event.inputs.app-path }}"
+          registry-password: ${{ secrets.GITHUB_TOKEN }}
+          artifact-name: ${{ needs.release-maven.outputs.ARTIFACT_NAME }}
+          registry: ghcr.io
```
[the remark this answers is not in the document]
Comment (hupling) [reply .github/workflows/release-maven-image.yaml:48]:
```suggestion
```

File: .github/workflows/release-maven-image.yaml [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        id: create_release
+        uses: softprops/action-gh-release@v2
+        with:
+          tag_name: ${{needs.release-maven.outputs.MVN_ARTIFACT_ID}}-${{ github.event.inputs.releaseVersion }}
+          draft: false
+          prerelease: false
+          generate_release_notes: false
+          draft: false
+          prerelease: false
+          generate_release_notes: false
```
[the remark this answers is not in the document]
Comment (hupling) [reply .github/workflows/release-maven-image.yaml:67]:
```suggestion
```

File: .github/workflows/release-maven-image.yaml [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+        with:
+          app-path: "/${{ github.event.inputs.app-path }}"
+          releaseVersion: "${{ github.event.inputs.releaseVersion }}"
+          developmentVersion: "${{ github.event.inputs.developmentVersion }}"
+
+  build-image:
+    runs-on: ubuntu-latest
+    needs: release-maven
+    steps:
+      - uses: it-at-m/.github/.github/actions/action-build-image@main
```
Comment (hupling) [inline .github/workflows/release-maven-image.yaml:42]: vielleicht kann man hier noch skip deployment hinzufügen. Dann hat man alles von hier kopiert https://git.muenchen.de/ccse/ospo/-/wikis/Maven-Central#releaseprozess-in-github-actions

File: .github/workflows/release-maven-image.yaml [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Create GitHub Release
+        id: create_release
+        uses: softprops/action-gh-release@v2
+        with:
+          tag_name: ${{needs.release-maven.outputs.MVN_ARTIFACT_ID}}-${{ github.event.inputs.releaseVersion }}
+          draft: false
+          prerelease: false
+          generate_release_notes: false
```
[the remark this answers is not in the document]
Comment (hupling) [reply .github/workflows/release-maven-image.yaml:63]: see above https://github.com/it-at-m/refarch-templates/pull/513#discussion_r1846428195

File: .github/workflows/release-maven-image.yaml
Comment (hupling) [reply .github/workflows/release-maven-image.yaml:1]: You don't want to create a jar File and than trigger the pipeline again and create an image and than trigger the pipeline to create a github release. Everything belongs together.

We have collected a lot of experience to know what is the best for out customers. You fog a lot of code so the user don't know what happend.

**Comment by hupling** [comment]
@ejcsid has the same opinion as me

File: .github/workflows/release-maven-image.yaml (hunk 2 above)
Comment (hupling) [reply .github/workflows/release-maven-image.yaml:67]: eingebaut

File: .github/workflows/release-maven-image.yaml (hunk 6 above)
Comment (hupling) [reply .github/workflows/release-maven-image.yaml:46]: eingebaut

**Comment by hupling** [comment]
@devtobi  I would do the wrapper in a separate issue because you have in other workflows  the same problem https://github.com/it-at-m/refarch-templates/blob/43-actions-to-create-releases/.github/workflows/maven-node-build.yaml#L23

I would do all together. so this mr is ready to merge

File: .github/workflows/release-maven-image.yaml (hunk 7 above)
[the remark this answers is not in the document]
Comment (hupling) [reply .github/workflows/release-maven-image.yaml:71]: seperate issue

File: .github/workflows/release-maven-image.yaml [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+          GDP_PRIVATE_KEY: ${{  secrets.gpg_private_key }}
+
+  build-image:
+    runs-on: ubuntu-latest
+    needs: release-maven
+    steps:
+      - uses: it-at-m/.github/.github/actions/action-build-image@main
+        with:
+          path: "${{ github.event.inputs.app-path }}"
+          image-name: "${{ github.event.inputs.app-path }}"
```
[the remark this answers is not in the document]
Comment (hupling) [reply .github/workflows/release-maven-image.yaml:49]: doesnt make sense

File: .github/workflows/release-maven-image.yaml [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+      - name: Create GitHub Release
+        id: create_release
+        uses: softprops/action-gh-release@v2
+        with:
+          tag_name: ${{needs.release-maven.outputs.MVN_ARTIFACT_ID}}-${{ github.event.inputs.releaseVersion }}
+          draft: false
+          prerelease: false
+          generate_release_notes: false
+          files: |
+            ${{ github.event.inputs.app-path }}/target/*.jar
```
[the remark this answers is not in the document]
Comment (hupling) [reply .github/workflows/release-maven-image.yaml:73]: if someone generates a .war file it is easier to understand

File: .github/workflows/release-maven-image.yaml [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+  workflow_dispatch:
+    inputs:
+      releaseVersion:
+        description: "Default version to use when preparing a release."
+        required: true
+        default: "X.Y.Z"
+      developmentVersion:
+        description: "Default version to use for new local working copy."
+        required: true
+        default: "X.Y.Z-SNAPSHOT"
```
[the remark this answers is not in the document]
Comment (hupling) [reply .github/workflows/release-maven-image.yaml:13]: doen't make sense for me, but @FabianWilms liked it so i will include it

**Review by DanielOber — APPROVED** [review]
LGTM for now, further improvements and ideas will be resolved in separate issues

**Comment by hupling** [comment]
https://github.com/it-at-m/.github/issues/82 hier das separate issue

**Pull request merged**